tmexp/train_hdp.py

```diff
@@ -0,0 +1,120 @@
+import os
+from typing import Dict, List, Tuple
+
+import gensim
+import numpy as np
+import tqdm
+
+from .create_bow import DOCWORD_FILE_NAME, VOCAB_FILE_NAME
+from .utils import check_exists, check_remove_file, create_directory, create_logger
+
+
+DOCTOPIC_FILE_NAME = "doc.topic.txt"
+WORDTOPIC_FILENAME = "word.topic.npy"
+
+
+def train_hdp(
+    input_dir: str,
+    output_dir: str,
+    dataset_name: str,
+    force: bool,
+    chunk_size: int,
+    kappa: float,
+    tau: float,
+    K: int,
+    T: int,
+    alpha: int,
+    gamma: int,
+    eta: float,
+    scale: float,
+    var_converge: float,
+    min_proba: float,
+    log_level: str,
+) -> None:
+    logger = create_logger(log_level, __name__)
+
+    input_dir = os.path.join(input_dir, dataset_name)
+    output_dir = os.path.join(output_dir, dataset_name)
+    words_input_path = os.path.join(input_dir, VOCAB_FILE_NAME)
+    check_exists(words_input_path)
+    docword_input_path = os.path.join(input_dir, DOCWORD_FILE_NAME)
+    check_exists(docword_input_path)
+    doctopic_output_path = os.path.join(output_dir, DOCTOPIC_FILE_NAME)
+    check_remove_file(doctopic_output_path, logger, force)
+    wordtopic_output_path = os.path.join(output_dir, WORDTOPIC_FILENAME)
+    check_remove_file(wordtopic_output_path, logger, force)
+    create_directory(output_dir, logger)
+
+    logger.info("Loading bags of words ...")
+    with open(docword_input_path, "r") as fin:
+        corpus: List[List[Tuple[int, int]]] = [[] for _ in range(int(fin.readline()))]
+        logger.info("Number of documents: %d", len(corpus))
+        num_words = int(fin.readline())
+        logger.info("Number of words: %d", num_words)
+        num_rows = int(fin.readline())
+        logger.info("Number of document/word pairs: %d", num_rows)
+        for line in tqdm.tqdm(fin.readlines(), total=num_rows):
+            doc_id, word_id, count = line.split()
+            corpus[int(doc_id)].append((int(word_id), int(count)))
+    logger.info("Corpus created.")
+
+    logger.info("Loading vocabulary ...")
+    with open(words_input_path, "r") as fin:
+        word_index: Dict[int, str] = {
```

Comment:
is the type annotation necessary? Usually mypy complains only for empty collections. I'm fine to keep it either way, as you prefer

Reply:
I think it was, not a big deal to keep imo